### PR TITLE
Added user details service, configured authentication and roles

### PR DESCRIPTION
**Pull request by frostpv**

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/RoleDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            }
+        }
+    }
+
+    @Override
+    public Role getRoleByName(String roleName) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<Role> userQuery =
+                    session.createQuery("from Role where RoleName = :name", Role.class);
+            userQuery.setParameter("name", roleName);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/RoleDaoImpl.java]: does this method work correctly?

File: src/main/java/com/dev/cinema/security/CustomUserDetailsService.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) {
+        com.dev.cinema.model.User user = userService.findByEmail(email).get();
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/security/CustomUserDetailsService.java]:
```suggestion
        com.dev.cinema.model.User user = userService.findByEmail(email).orElseThrow(() -> new UsernameNotFoundException(message));
```

**Review by Sofasmile — COMMENTED** [review]
le'ts add Role USER to user in authentication service register() method

File: src/main/java/com/dev/cinema/dao/RoleDaoImpl.java (hunk 1 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/dao/RoleDaoImpl.java]: fixed

File: src/main/java/com/dev/cinema/security/CustomUserDetailsService.java (hunk 2 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/security/CustomUserDetailsService.java]: fixed

**Review by Sofasmile — COMMENTED** [review]
how about this comment?
image

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**